save-backend/src/main/kotlin/com/saveourtool/save/backend/security/TestSuitePermissionEvaluator.kt

```diff
@@ -1,25 +1,42 @@
 package com.saveourtool.save.backend.security
 
+import com.saveourtool.save.backend.service.LnkOrganizationTestSuiteService
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.Organization
 import com.saveourtool.save.entities.TestSuite
+import com.saveourtool.save.permission.Permission
+import com.saveourtool.save.permission.Rights
 import org.springframework.security.core.Authentication
 import org.springframework.stereotype.Component
 
 /**
- * Class that is capable of assessing user's permissions regarding test suites.
+ * Class that is capable of assessing organization's permissions regarding test suites.
  */
 @Component
-class TestSuitePermissionEvaluator {
+class TestSuitePermissionEvaluator(
+    private var lnkOrganizationTestSuiteService: LnkOrganizationTestSuiteService
+) {
     /**
-     * fixme: just a remark for future work
-     *
-     * @param testSuite test suite that is being requested
+     * @param organization
+     * @param testSuite
+     * @param permission
      * @param authentication
-     * @return true if user with [authentication] can access [testSuite], false otherwise
+     * @return true if [organization] has required [permission] over [testSuite]
      */
-    fun canAccessTestSuite(
+    fun hasPermission(
+        organization: Organization,
         testSuite: TestSuite,
-        authentication: Authentication?
-    ) = authentication?.let {
-        true
-    } ?: false
+        permission: Permission,
+        authentication: Authentication?,
+    ): Boolean = lnkOrganizationTestSuiteService.getDto(organization, testSuite).rights.let { currentRights ->
+        authentication?.hasRole(Role.SUPER_ADMIN) == true || when (permission) {
+            Permission.READ -> testSuite.isPublic || canAccessTestSuite(currentRights)
+            Permission.WRITE, Permission.DELETE -> canMaintainTestSuite(currentRights)
+        }
+    }
+
+    private fun canMaintainTestSuite(rights: Rights) = rights == Rights.MAINTAIN
+    private fun canAccessTestSuite(rights: Rights) = canMaintainTestSuite(rights) || rights == Rights.USE
+
+    private fun Authentication.hasRole(role: Role): Boolean = authorities.any { it.authority == role.asSpringSecurityRole() }
```

Comment:
it already presents in three places, maybe move to SpringUtils ?

Reply:
Nice idea, thx, will move it

Reply:
Done